docs/user-namespaces/README.md

```diff
@@ -0,0 +1,146 @@
+# Support for user namespaces
+
+Kubernetes supports running pods with user namespace since v1.25. This document explains the
+containerd support for this feature.
+
+## What are user namespaces?
+
+A user namespace isolates the user running inside the container from the one in the host.
+
+A process running as root in a container can run as a different (non-root) user in the host; in
+other words, the process has full privileges for operations inside the user namespace, but is
+unprivileged for operations outside the namespace.
+
+You can use this feature to reduce the damage a compromised container can do to the host or other
+pods in the same node. There are several security vulnerabilities rated either HIGH or CRITICAL that
+were not exploitable when user namespaces is active. It is expected user namespace will mitigate
+some future vulnerabilities too.
+
+See [the kubernetes documentation][kube-intro] for a high-level introduction to
+user namespaces.
+
+[kube-intro]: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#introduction
+
+## Stack requirements
+
+The Kubernetes implementation was redesigned in 1.27, so the requirements are different for versions
+pre and post Kubernetes 1.27.
+
+Please note that if you try to use user namespaces with containerd 1.6 or older, the `hostUsers:
+false` setting in your pod.spec will be **silently ignored**.
+
+### Kubernetes 1.25 and 1.26
+
+ * Containerd 1.7 or greater
+ * runc 1.1 or greater
+
+### Kubernetes 1.27 and greater
+
+ * Linux 6.3 or greater
+ * Containerd 2.0 or greater
+ * You can use runc or crun as the OCI runtime:
+   * runc 1.2 or greater
+   * crun 1.9 or greater
+
+Furthermore, all the file-systems used by the volumes in the pod need kernel-support for idmap
+mounts. Some popular file-systems that support idmap mounts in Linux 6.3 are: `btrfs`, `ext4`, `xfs`,
+`fat`, `tmpfs`, `overlayfs`.
+
+The kubelet is in charge of populating some files to the containers (like configmap, secrets, etc.).
+The file-system used in that path needs to support idmap mounts too. See [the Kubernetes
+documentation][kube-userns-req] for more info on that.
+
+
+[kube-req]: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#before-you-begin
```

Comment:
small nit: typo in that the ref and the ref name+link do not match, so this link isn't being rendered. [kube-userns-req] vs. [kube-req]

Reply:
Oh, thanks! Fixed now